shared/ghutil/issue.go

```diff
@@ -63,20 +63,19 @@ func (gc *GithubClient) ListRepos(org string) ([]string, error) {
 
 // ListIssuesByRepo lists issues within given repo, filters by labels if provided
 func (gc *GithubClient) ListIssuesByRepo(org, repo string, labels []string) ([]*github.Issue, error) {
-	issueListOptions := github.IssueListByRepoOptions{
+	issueListOptions := &github.IssueListByRepoOptions{
 		State: string(IssueAllState),
 	}
 	if len(labels) > 0 {
 		issueListOptions.Labels = labels
 	}
 
-	options := &github.ListOptions{}
 	genericList, err := gc.depaginate(
 		fmt.Sprintf("listing issues with label '%v'", labels),
 		maxRetryCount,
-		options,
+		&issueListOptions.ListOptions,
 		func() ([]interface{}, *github.Response, error) {
```

Comment:
I can't help thinking that you could simplify depaginate even more, by encapsulating this repeated pattern (call gc.Client/for/append/return) into it and just passing a function that calls gc.Client.X. But I won't block on it.

Reply:
I think you're absolutely right. I will let this PR in first, and see what I can do refactoring this in another PR

Reply:
I have done some investigation, the sad news is this is almost not possible, due to the lack of generic support in Go. As in this case, `ListByRepo` function can not be passed in as a  parameter as the arguments vary among other `List` functions. And this line `page, resp, err := gc.Client.Issues.ListByRepo(ctx, org, repo, issueListOptions)` can not be used as an argument, as the return type(`page`) is different among other `List` functions. So seems like converting `page` to an interface slice locally is not avoidable. @adrcunha

Reply:
:(